File: gaussint.py
'''plot the integrand and approximate
the integral 
    / 1
    |    exp(-x^2/pi) dx
    / 0
by left-hand, right-hand, and
trapezoid rules'''

import numpy as np
import matplotlib.pyplot as plt
from scipy.special import erf

N = 1000
dx = (1.0 - 0.0) / N
x = np.linspace(0.0,1.0,N+1)
y = np.exp(- x**2 / np.pi)

plt.plot(x,y)
plt.axis([0.0,1.0,0.0,1.0])
plt.grid(True)

lhand = dx * sum(y[:-1])
print("lhand = %.15f" % lhand)
rhand = dx * sum(y[1:])
print("rhand = %.15f" % rhand)
trap  = (dx/2) * sum(y[:-1]+y[1:])
print("trap = %.15f" % trap)
exact = (np.pi/2) * erf(1/np.sqrt(np.pi))
print("exact = %.15f" % exact)
plt.show()

File: gaussint.jl
# plot the integrand and approximate
# the integral 
#   / 1
#   |    exp(-x^2/pi) dx
#   / 0
# by left-hand, right-hand, and
# trapezoid rules
# example:
#   julia> include("gaussint.jl")

using Plots
using Printf
using SpecialFunctions

N = 1000
dx = (1 - 0) / N;
x = range(0, 1, length=N+1)
y = exp.(- x.^2 / pi)

display(plot(x, y))

lhand = dx * sum(y[1:end-1])
@printf("lhand = %.15f\n", lhand)
rhand = dx * sum(y[2:end])
@printf("rhand = %.15f\n", rhand)
trap  = (dx/2) * sum(y[1:end-1] + y[2:end])
@printf("trap = %.15f\n", trap)
exact = (pi/2) * erf(1/sqrt(pi))
@printf("exact = %.15f\n", exact)

println("Press Enter to continue...")
readline()
